### DISCUSSION TITLE
Can Next-Auth support multi-email account linking?

### DISCUSSION THREAD

**Discussion by nbouvrette** [nextauthjs/next-auth#6162] [Help] [answered]
We just refactored our next-auth implementation and it works nicely across different providers using `allowDangerousEmailAccountLinking: true`.

Having said this, we are looking to mimic Facebook's behavior and allow multiple emails (even maybe providers?) on the same account.

Of course, adding and linking accounts would be done on our side (I don't think it's next-auth's responsibility). But if we link 2 different email addresses on the same account, and that specific account tries to log in on a new provider using one of the verified email addresses, will `allowDangerousEmailAccountLinking: true` still work? 

I couldn't find too many details on how this logic works and if it would work with more than 1 email.

**Comment by nbouvrette** [comment, marked as answer]
I found the answer here https://github.com/nextauthjs/next-auth/discussions/3126 based on this discussion https://github.com/nextauthjs/next-auth/discussions/990#discussioncomment-241702

To summarize:

- email addresses are only used on the `user` table during the original sign up
- you can link accounts to different providers by allowing a logged-in user to "sign-in" to another provider.
  - when linking other accounts, the email address is not updated
- to unlink providers I am not sure how it will work yet but I presume we might need to do this manually on the backend since I saw a few next-auth documents mention this was "to be implemented"

Supporting multiple emails is not supported by the next-auth data model but could probably be supported by extending it.

**Reply by stimw** [reply]
This definitely should be on the documents..I spent a long time looking for answers in documents and discussions.
